lib/cfg.py

```diff
@@ -251,6 +253,83 @@ def cfg_version(filename):
         cfg[section] = d
     return cfg
 
+def cfg_altiumWorkspace(options):
+    """
+    ricerca del nome di tutti i progetti all'interno del file Workspace
+    """
+    risultato=True
+    path_dict = {}
+    config = ConfigParser.RawConfigParser()
+    config.read(options.ws)
+
+    for i in config.sections():
+        try:
+            temp = config.get(i, 'ProjectPath')
+            temp = temp.split('/')
+            k = []
+            p = []
+
+            if len(temp) > 1:
+                k = os.path.splitext(temp[1])[0]
+                p = os.path.join(*temp)
+            else:
+                k = os.path.splitext(temp[0])[0]
+                p = os.path.join("./", *temp)
+
+            if len(k) == 0 or len(p) == 0:
+                lib.error("Errore nel parsing del workspace: %s" % e,
+                            self.handler, terminal=self.terminal)
+                risultato =False
+
+            k = k.lower()
+            path_dict[k] = p
+
+        except ConfigParser.NoOptionError:
+            pass
+    path=''
+    ws=options.ws.split('/')
+    for i in range(0, (len(ws)-1)):
```

Comment:
Apprezzo la fantasia, ma in python si può ciclare sulle liste senza dovere usare fantasiosi indici..
```
for i in lista_di_oggeti:
    print i
```